V1.0

BrandOtter opened the pull request:
Discuss changes made to reflect backend development

anbroussard committed:
I added some temp code to the updateEnrollmentGrade function in the EnrollmentController. The EnrollmentRepository file was also updated; some code was added to the updateEnrollmentGrade function becuase I was getting an error when trying to run the application.

anbroussard committed:
Created variables and Setter/Getters for Grade.java. Also uncommented the Query in GradeRepository.java.

BrandOtter committed:
EnrollmentController Update

Co-Authored-By: maleman <snax2251@gmail.com>

BrandOtter committed:
Merge branch 'v1.0' of https://github.com/BrandOtter/cst438_Assignment2-05 into v1.0

BrandOtter committed:
Revert "EnrollmentController Update"

This reverts commit a67cbb8bdb1b23b203bafb1a4bd82eccc4a3d729.

anbroussard committed:
Added variables/Getters/Setters to Assignment.java. Also Uncommented queries in AssignmentRepository.java. Again, I had to add value and nativeQuery assignments to the second query because it was triggering errors when running in VS Code. These assignments can be removed if no one else is getting this error.

BrandOtter committed:
Completed EnrollmentController and Updated StudentController

Co-Authored-By: mealeman <116986635+mealeman@users.noreply.github.com>

BrandOtter committed:
Merge branch 'v1.0' of https://github.com/BrandOtter/cst438_Assignment2-05 into v1.0

BrandOtter committed:
Update StudentController.java

Completed work on Student Controller. Pushed to test and confirm correct functionality.

Co-Authored-By: mealeman <116986635+mealeman@users.noreply.github.com>

anbroussard committed:
Some adjustments to Assignment files flesh out the getMapping for Assignment.

BrandOtter committed:
Fixed EnrollmentRepository

Added necessary code for SpringBoot to register and accept the queries within the EnrollmentRepository

BrandOtter committed:
Merge branch 'v1.0' of https://github.com/BrandOtter/cst438_Assignment2-05 into v1.0

BrandOtter committed:
Update StudentController.java

Added idempotency check for the dropCourse method

BrandOtter committed:
Update AssignmentController.java

Assignment controller methods update

BrandOtter committed:
Completed AssignmentController Methods

All complete, except for the check about whether the current user is an instructor. Will revisit.

BrandOtter reviewed (commented):
Might be good to add comments in order to help identify what some of the code is intending to do (note to self).

Pekoe200 commented:
this upcoming sprint we should do unit tests first, then the code.

anbroussard commented:
Agreed with the point about comments. I removed the initial comments supplied with the code when I wrote my functions. It would have been helpful to leave updated comments for how new code worked. 

Unit tests will be tremendously helpful to testing the functionality of the front end this coming week.

anbroussard commented on src/main/java/com/cst438/controller/EnrollmentController.java:
The edit to the exception code for the UpdateEnrollmentGrade function is much more optimized than the previous implementation.

mealeman commented:
Unit tests will be helpful going forward, for now things seem to be working

cmcmichael-csumb commented on src/main/java/com/cst438/controller/AssignmentController.java:
```diff
@@ -3,50 +3,95 @@
-        return null;
+        List<Assignment> assignments = assignmentRepository.findBySectionNoOrderByDueDate(secNo);
+
+        if(assignments == null){
+            throw  new ResponseStatusException( HttpStatus.NOT_FOUND, "section number is invalid");
+        }
```
From my testing, this error statement is never ran no matter what section number is used. Tested (0, 5, 8, 10, 20)

cmcmichael-csumb commented on src/main/java/com/cst438/controller/AssignmentController.java:
```diff
@@ -55,64 +100,118 @@ public AssignmentDTO createAssignment(
+        Assignment assignment = assignmentRepository.findById(assignmentId)
+                .orElseThrow(() -> new ResponseStatusException(HttpStatus.NOT_FOUND, "Assignment not found"));
+
+        if (assignment.getSection().getSecId() != dto.secId()) {
+            Section section = sectionRepository.findById(dto.secId())
+                    .orElseThrow(() -> new ResponseStatusException(HttpStatus.NOT_FOUND, "Section not found"));
+            assignment.setSection(section);
+        }
```
See line 97, this seems to allow the updating of an assignments section, which isn't an intended feature.

cmcmichael-csumb commented on src/main/java/com/cst438/controller/EnrollmentController.java:
```diff
@@ -39,6 +38,11 @@ public List<EnrollmentDTO> getEnrollments(
 
         List<Enrollment> enrollments = enrollmentRepository.findEnrollmentsBySectionNoOrderByStudentName(sectionNo);
+
+        if(enrollments == null){
+            throw  new ResponseStatusException( HttpStatus.NOT_FOUND, "section number is invalid");
+        }
```
Same idea as in AssignmentController, I don't believe this error state ever runs.

cmcmichael-csumb reviewed (approved):
JPA entity for Enrollment is completed -> 5
JPA entity for Assignment is completed -> 5
JPA entity for Grade is completed -> 5
RestController StudentController is completed -> 5
RestController AssignmentController is completed -> 5
RestController EnrollmentController is completed -> 5

BrandOtter committed:
Corrections to Repository Files

Suggestions made by IntelliJ caused the queries to work incorrectly. Queries have been fixed and should now get data back properly.

Pekoe200 committed:
Added unit tests for AssignmentController

BrandOtter committed:
AssignmentId DTO Fix

Fixed an issue where AssignmentId was not correctly serializing. Working now.

anbroussard committed:
Created StuddentControllerUnitTest file. Made functionality changes to the Student contoller. Added a New course to the data for testing.

anbroussard committed:
updated StudentControllerUnitTest with the rest of the enrollment unit tests. HW items 6-9 are completed. Added an additional test for deleting an enrollment past the drop deadline.

mealeman committed:
added failGradeAssignment test and added @Test where needed

Pekoe200 committed:
Added test for due date past term date

Pekoe200 committed:
fixed merge issues on AssigmentControllerUnitTest

anbroussard committed:
Updated the StudentControllerUnit with HW item #10. Had trouble to the verification tests. Also added an enrollment item to the data.

anbroussard committed:
Merge branch 'v1.0' of github.com:BrandOtter/cst438_Assignment2-05 into v1.0

BrandOtter committed:
Added Selenium Tests p1

Added first two Selenium tests, and configured environment to work with driver on native computer

BrandOtter committed:
Merge branch 'v1.0' of https://github.com/BrandOtter/cst438_Assignment2-05 into v1.0

anbroussard committed:
Merge branch 'v1.0' of github.com:BrandOtter/cst438_Assignment2-05 into v1.0

BrandOtter committed:
Test Case Correction

Fixed Selenium test cases to be more consistent and have appropriate end routines

anthonymatricia commented:
Unit Tests look good. System Tests look good. Nicely done!

BrandOtter committed:
Separated out System Tests

Tests were all nested under the SectionControllerSystemTest.java file. Have been split out and given their own drivers to run Selenium independently.

BrandOtter committed:
Corrected some System Tests

Added better descriptions, and fixed the state of one not being correct
